src/js/components/ProductManageMenu/ProductManageMenu.js

```diff
@@ -0,0 +1,134 @@
+import { SELECTOR } from '../../constants/selector.js';
+import { $ } from '../../utils/dom.js';
+import { EmptyInputError, InvalidValueError } from '../../utils/error.js';
+import { productStorage } from '../../utils/storage.js';
+import { validateProductName, validateProductPrice, validateProductQuantity } from '../../utils/validation.js';
+
+/* eslint-disable class-methods-use-this */
+export default class ProductManageMenu {
+  #state = {
+    name: '',
+    price: '',
+    quantity: '',
+    products: new Map(),
+  };
+
+  get state() {
+    return this.#state;
+  }
+
+  set state(state) {
+    this.#state = state;
+  }
+
+  init() {
+    this.#state = this.#getInitialState();
+    this.#render();
+    this.#bindEvents();
+  }
+
+  #getInitialState() {
+    return {
+      products: productStorage.get('products'),
+    };
+  }
+
+  #resetInput() {
+    $(SELECTOR.PRODUCT_NAME_INPUT).value = '';
+    $(SELECTOR.PRODUCT_PRICE_INPUT).value = '';
+    $(SELECTOR.PRODUCT_QUANTITY_INPUT).value = '';
+  }
+
+  #addProduct(product) {
+    this.#state.products.set(product.name, product);
+    productStorage.set('products', this.#state.products);
+    this.#render();
+    this.#bindEvents();
+  }
+
+  #handleChangeName(e) {
+    this.#state.name = e.target.value;
+  }
+
+  #handleChangePrice(e) {
+    this.#state.price = e.target.valueAsNumber;
+  }
+
+  #handleChangeQuantity(e) {
+    this.#state.quantity = e.target.valueAsNumber;
+  }
+
+  #handleProductAddButtonClick() {
+    try {
+      const { name, price, quantity } = this.#state;
+
+      validateProductName(name);
+
+      validateProductPrice(price);
+
+      validateProductQuantity(quantity);
+
+      const product = { name, price, quantity };
+
+      this.#addProduct(product);
+      this.#resetInput();
+    } catch (error) {
+      if (error instanceof EmptyInputError) {
+        alert(error.message);
+      }
+      if (error instanceof InvalidValueError) {
+        alert(error.message);
+      }
```

Comment:
`InvalidValueError`와 `EmptyInputError`가 모두 별도로 `CustomError`를 상속받으면 이렇게도 가능할 것 같아요
```suggestion
      if (error instanceof CusttomError) {
        alert(error.message);
      }
```

Reply:
instanceof가 프로토타입 체인을 타고 올라가면서 확인을 하는데 
최상위엔 CustomError 클래스의 프로토타입이 있을테니 훨씬 깔끔하니 좋은 것 같습니다!